docs-2.0/1.introduction/3.nebula-graph-architecture/4.storage-service.md

```diff
@@ -1,11 +1,261 @@
+The other is Storage Service that stores the data, which is run by nebula-storaged process. This topic will describe the architecture of Storage Service.
 
-    References:
+## Advantages
 
-    * [An Introduction to Nebula Graph's Storage Engine](https://nebula-graph.io/posts/nebula-graph-storage-engine-overview/)
-    * [Architecture overview](1.architecture-overview.md)
-    * [Meta Service](2.meta-service.md)
+- High performance (independent R&D KVStore)
+
+- Great scalability (Shared-nothing architecture, independent of hardware like NAS)
+
+- Strong consistency (Raft)
+
+- High availability (Raft)
+
+- Supports synchronizing with the third party systems, such as [Elasticsearch](../4.deployment-and-installation/6.deploy-text-based-index/2.deploy-es.md).
+
+## The architecture of Storage Service
+
+![image](https://docs-cdn.nebula-graph.com.cn/1.overview/3.design-and-architecture/2.storage-design/storage-design.png)
+
+Storage Service is run by nebula-storaged process. Users can deploy nebula-storaged processes on different occasions. For example, users can deploy 1 nebula-storaged process in test environment while deploy 3 nebula-storaged processes in production environment.
+
+All the nebula-storaged processes consist the Raft-based cluster. There are three layers in the Storage Service.:
+
+- Storage interface
+
+   The top layer is storage interface. It defines a set of APIs that are related to the graph. These API requirements will be translated into a set of KV operations targeting the corresponding [Partition](#data_partitioning). For example:
+
+  - `getNeighbors`: query the in-edge or out-edge of a set of vertices, return the edges and the corresponding properties, and support conditional filters.
+
+  - `insert vertex/edge`：insert a vertex or edge and its properties.
+
+  - `getProps`：get the properties of a vertex or an edge.
+
+  It is this layer that makes the Storage Service a real graph storage. Otherwise it is just a KV storage.
+
+- Consensus
+
+  Below the storage interface is the consensus layer that implements [Multi Group Raft](#multi_group_raft), which ensures the strong consistency and high availability of Storage Service.
+
+- Store engine
+
+  The bottom layer is the local storage engine, providing operations like `get`, `put`, and `scan` on local data. The related interfaces are stored in `KVStore.h` and `KVEngine.h` files. Users can develop their own local store plugins based on their needs.
+
+The following will describe some features of Storage Service based on the above architecture.
+
+## KVStore
+
+Nebula Graph develops its own KVStore to meet the performance needs of:
+
+- High performance KVStore.
+
+- Library-formed storage service for efficient pushdown. As a strong-typed database, how to provide Schema during pushdown is the key of efficiency.
```

Comment:
it is provided as a (kv) library and can be easily developed for the filtering-pushdown purpose. As a strong-typed database, how to get schema during the filtering-pushdown is the key to efficiency.